pandas/core/computation/eval.py

```diff
@@ -232,8 +237,13 @@ def eval(expr, parser='pandas', engine=None, truediv=True,
     pandas.DataFrame.query
     pandas.DataFrame.eval
     """
-    inplace = validate_bool_kwarg(inplace, 'inplace')
-    first_expr = True
+
+    inplace = validate_bool_kwarg(inplace, "inplace")
+    modifiable = hasattr(target, "__setitem__")
+
+    if inplace and not modifiable:
+        raise ValueError("Cannot modify the provided target inplace")
```

Comment:
Another possibility is to just raise depending on `parsed_expr.assigner`. If there is no assignment in the expression, raise when inplace is set to True